### PR TITLE
Moving MarkDuplicates transform into it's own file.

### PR DESCRIPTION
**Pull request by davidadamsphd**
This is a strict copy-paste job. I'll do further refactoring after this. I'm doing this in two steps so it's easier to read the diffs.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java [hunk 1]
```diff
@@ -71,401 +72,20 @@ protected void setupPipeline(final Pipeline pipeline) {
...
-
-    /**
      * Returns the list of intervals from the given sequence dictionary.
      */
     private static List<SimpleInterval> getAllIntervalsForReference(final SAMSequenceDictionary sequenceDictionary) {
-    return GenomeLocSortedSet.createSetFromSequenceDictionary(sequenceDictionary)
-            .stream()
-            .map(SimpleInterval::new)
-            .collect(Collectors.toList());
+        return GenomeLocSortedSet.createSetFromSequenceDictionary(sequenceDictionary)
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java]: replace with `IntervalUtils.getAllIntervalsForReference`

**Comment by lbergelson** [comment]
@davidadamsphd 1 comment.  Either address that now or in your next refactoring commits.  Otherwise :+1: merge at will.

**Comment by tovanadler** [comment]
I have a pull request out for this code. Will you refactoring maybe that
obsolete?

**Comment by lbergelson** [comment]
@tovanadler It's likely that your pull request will have to be rebased on the new code.

**Comment by droazen** [comment]
@tovanadler I believe the refactoring in question will be minimal (only what is necessary to connect `MarkDuplicatesDataflow` to the `ReadsPreprocessingPipeline`), so the conflict with your branch shouldn't be too bad. If you need a git consultant to assist with the rebase, we're glad to help :)

File: src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/MarkDuplicates.java [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * The main trasform for MarkDuplicates. Takes reads, returned reads with some reads marked as duplicates.
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/MarkDuplicates.java]: trasform -> transform
returned -> returns

Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/MarkDuplicates.java]: done.

File: src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java (hunk 1 above)
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java]: Done.

**Pull request merged**